DatadogRUM/Sources/SDKMetrics/SessionEndedMetricController.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// A controller responsible for managing "RUM Session Ended" metrics.
+internal final class SessionEndedMetricController {
+    /// Dictionary to keep track of pending metrics, keyed by session ID.
+    @ReadWriteLock
+    private var metricsBySessionID: [RUMUUID: SessionEndedMetric] = [:]
+    /// Array to keep track of pending session IDs in their start order.
+    @ReadWriteLock
+    private var pendingSessionIDs: [RUMUUID] = []
```

Comment:
/suggestion As this is private, and tightly linked to `metricsBySessionID`, it can be mutated while acquiring the lock of `metricsBySessionID`. Starting a metric would look like:
```swift
_metricsBySessionID.mutate { metrics in
    metrics[sessionID] = SessionEndedMetric(sessionID: sessionID, precondition: precondition, context: context)
    pendingSessionIDs.append(sessionID)
}
```

Reply:
> As this is private and tightly linked (...)

Fair call, done ✅